lighthouse-core/gather/gatherers/fonts.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * @license Copyright 2017 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Gatherer = require('./gatherer');
+const fontUrlRegex = new RegExp('url\\((?:"|\')([^"]+)(?:"|\')\\)');
+
+/* eslint-disable */
+function getAllLoadedFonts() {
+  const getFont = fontFace => ({
+    display: fontFace.display,
+    family: fontFace.family,
+    stretch: fontFace.stretch,
+    style: fontFace.style,
+    weight: fontFace.weight,
+  });
+
+  if (document.fonts.status === 'loaded') {
+    return Promise.resolve(
+      Array.from(document.fonts).filter(fontFace => fontFace.status === 'loaded')
+        .map(getFont)
+    );
+  } else {
+    return document.fonts.ready.then(() => {
+      return Array.from(document.fonts).filter(fontFace => fontFace.status === 'loaded')
+        .map(getFont);
+    });
+  }
+}
+function getFontFaceFromStylesheets() {
+  const fontFaceRules = [];
+  for (let sheet = 0; sheet < document.styleSheets.length; sheet++) {
+    for (var i = 0; document.styleSheets[sheet].cssRules && i < document.styleSheets[sheet].cssRules.length; i++) {
+      var rule = document.styleSheets[sheet].cssRules[i];
+
+      if (rule instanceof CSSFontFaceRule) {
+        const keys = Object.keys(rule.style);
```

Comment:
feels a bit odd to iterate over the properties in this way, can we just pull the font data directly?

```
fontFaceRules.push({
  display: rule.style.fontDisplay,
  ...
})
```

Reply:
sure we can, just wanted this audit to be future proof if any more rules were added to font-face that we could use but maybe just add it if when we need it

Reply:
ah, I'd be in favor of clearer code until we need actually more font props

not sure about others

Reply:
I follow your reasoning so 👍